src/org/ods/orchestration/usecase/BitbucketTraceabilityUseCase.groovy

```diff
@@ -0,0 +1,221 @@
+import org.ods.services.BitbucketService
+import org.ods.util.IPipelineSteps
+
+import java.text.SimpleDateFormat
+
+@groovy.lang.Grab('com.xlson.groovycsv:groovycsv:1.3')
+import com.xlson.groovycsv.CsvParser
+
+class BitbucketTraceabilityUseCase {
+
+    private static final String CSV_FILE = "source-code-review.csv"
+    private static final int PAGE_LIMIT = 10
+
+    private final BitbucketService bitbucketService
+    private final IPipelineSteps steps
+    private final Project project
+
+    BitbucketTraceabilityUseCase(BitbucketService bitbucketService, IPipelineSteps steps, Project project) {
+        this.steps = steps
+        this.project = project
+        this.bitbucketService = bitbucketService
+    }
+
+    /**
+     * Create a CSV file that contains the following records
+     * for every merge event into the integration branch of every ODS component:
+     * @return absolutePath of the created file
+     */
+    @SuppressWarnings(['JavaIoPackageAccess'])
+    String generateSourceCodeReviewFile() {
+        def file = new File("${steps.env.WORKSPACE}/${CSV_FILE}")
+        if (file.exists()) {
+            file.delete()
+        }
+        file.createNewFile()
+
+        def token = bitbucketService.getToken()
+        List<Map> repos = getRepositories()
+        repos.each {
+            processRepo(token, it, file)
+        }
+        return file.absolutePath
+    }
+
+
+    /**
+     * Read an existing csv file and parse the info to obtaining an structured List of data.
+     *
+     * @param filePath The csv
+     * @return List of commits
+     */
+    @SuppressWarnings(['JavaIoPackageAccess'])
+    List<Map> readSourceCodeReviewFile(String filePath) {
+        def file = new File(filePath)
+        def result = []
+        def data = processCsv(file.text, Record.CSV,
+            ['commitDate', 'author', 'reviewers', 'pullRequestUrl', 'commitSHA', 'component'])
+
+        for (info in data) {
+            def authorInfo = processCsvDeveloper(info.author)
+            def reviewers = []
+            info.reviewers.split(Record.REVIEWERS_DELIMITER).each {
+                def reviewerInfo = processCsvDeveloper(it)
+                reviewers << [reviewerName: reviewerInfo.name, reviewerEmail: reviewerInfo.email]
+            }
+
+            def commitInfo = [
+                date: info.commitDate,
+                authorName: authorInfo.name,
+                authorEmail: authorInfo.email,
+                reviewers: reviewers,
+                url: info.pullRequestUrl,
+                commit: info.commitSHA,
+                component: info.component,
+            ]
+            result << commitInfo
+        }
+
+        return result
+    }
+
+    private void processRepo(String token, Map repo, File file) {
+        def nextPage = true
+        def nextPageStart = 0
+        while (nextPage) {
+            def commits = bitbucketService.getCommitsForMainBranch(token, repo.repo, PAGE_LIMIT, nextPageStart)
```

Comment:
As a minor improvement, I suggest to change the method name from *Main* into *Integration*, since `repo.repo` names the integration branch.

Reply:
Changed to: _getCommitsForIntegrationBranch_